matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/keysbackup/KeysBackup.kt

```diff
@@ -601,61 +615,63 @@ class KeysBackup(private val mCrypto: MXCrypto, session: MXSession) {
+                            Log.d(LOG_TAG, "checkAndStartKeysBackup: Found usable key backup. version: " + keyBackupVersion.version)
                             when {
                                 mKeysBackupVersion == null -> {
                                     // Check the version we used at the previous app run
                                     val versionInStore = mCrypto.cryptoStore.keyBackupVersion
                                     if (versionInStore != null && versionInStore != keyBackupVersion.version) {
                                         Log.d(LOG_TAG, " -> clean the previously used version $versionInStore")
-                                        disableKeyBackup()
+                                        disableKeysBackup()
                                     }
 
                                     Log.d(LOG_TAG, "   -> enabling key backups")
-                                    enableKeyBackup(keyBackupVersion)
+                                    enableKeysBackup(keyBackupVersion)
                                 }
                                 mKeysBackupVersion!!.version.equals(keyBackupVersion.version) -> {
                                     Log.d(LOG_TAG, "   -> same backup version (" + keyBackupVersion.version + "). Keep using it")
+                                    mKeysBackupStateManager.state = KeysBackupStateManager.KeysBackupState.ReadyToBackUp
                                 }
                                 else -> {
                                     Log.d(LOG_TAG, "   -> disable the current version (" + mKeysBackupVersion!!.version
                                             + ") and enabling the new one: " + keyBackupVersion.version)
-                                    disableKeyBackup()
-                                    enableKeyBackup(keyBackupVersion)
+                                    disableKeysBackup()
+                                    enableKeysBackup(keyBackupVersion)
                                 }
                             }
                         } else {
-                            Log.d(LOG_TAG, "checkAndStartKeyBackup: No usable key backup. version: " + keyBackupVersion.version)
+                            Log.d(LOG_TAG, "checkAndStartKeysBackup: No usable key backup. version: " + keyBackupVersion.version)
                             if (mKeysBackupVersion == null) {
                                 Log.d(LOG_TAG, "   -> not enabling key backup")
+                                mKeysBackupStateManager.state = KeysBackupStateManager.KeysBackupState.NotTrusted
                             } else {
                                 Log.d(LOG_TAG, "   -> disabling key backup")
-                                disableKeyBackup()
+                                disableKeysBackup()
+                                mKeysBackupStateManager.state = KeysBackupStateManager.KeysBackupState.Disabled
```

Comment:
Should be `NotTrusted` here too: There is backup on the homeserver but it is not usable.

Reply:
ok